src/libbson/src/bson/bson-decimal128.c

```diff
@@ -298,10 +298,12 @@ bson_decimal128_to_string (const bson_decimal128_t *dec, /* IN  */
             *(str_out++) = '0';
          }
 
-         const unsigned radix_stop = (unsigned) BSON_MAX (radix_position - 1, 0);
-         const unsigned n_to_write = significand_digits - radix_stop;
-         for (uint32_t i = 0; i < n_to_write && (str_out - str) < available_bytes; i++) {
-            *(str_out++) = *(significand_read++) + '0';
+         const unsigned dot_pos = (unsigned) BSON_MAX (radix_position - 1, 0);
+         const unsigned n_trailing_digits = num_significant_digits - dot_pos;
```

Comment:
```suggestion
         const unsigned n_trailing_digits = (unsigned) num_significant_digits - dot_pos;
```
With Clang 20.0.0:
```
warning: implicit conversion changes signedness: 'int32_t' (aka 'int') to 'unsigned int' [-Wsign-conversion]
```

Reply:
I'll leave this one but fix the other `irange`/`urange` ones, as I want to address most arithmetic-related `sign-conversion` warnings in a later change that adds ckdint APIs.